### ISSUE TITLE
[JENKINS-11962] Symlinking lastSuccessful build shouldn't fail with concurrent jobs

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-14#4903]
<p>I had three builds running at the same time, and two of them finished during the same second. This lead to the following message on one of them:</p>

<p><tt>ln -s builds/2011-12-01_21-35-22 /var/lib/jenkins/jobs/my_job/builds/../lastSuccessful failed: 17 File exists</tt></p>

<p>The job still succeeded, so it's not a big deal.</p>

<p>This seems like a race condition between rm-ing the old symlink and creating the new one. Maybe ln -sf would work better? I assume it does its operations atomically.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jorgenpt.png" title="jorgenpt's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jorgenpt">jorgenpt</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-11962" target="_blank">Symlinking lastSuccessful build shouldn't fail with concurrent jobs</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2023-06-23T00:57:44+00:00
<li><b>votes</b>: 4
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-14
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I had three builds running at the same time, and two of them finished during the same second. This lead to the following message on one of them:</p>

<p><tt>ln -s builds/2011-12-01_21-35-22 /var/lib/jenkins/jobs/my_job/builds/../lastSuccessful failed: 17 File exists</tt></p>

<p>The job still succeeded, so it's not a big deal.</p>

<p>This seems like a race condition between rm-ing the old symlink and creating the new one. Maybe ln -sf would work better? I assume it does its operations atomically.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 1.438<br/>
Concurrent job<br/>
3 builds running
```
</details>
<details><summary><i>5 attachments</i></summary>

- [ScreenShot.png](https://issues.jenkins.io/secure/attachment/27993/ScreenShot.png)
  > ![ScreenShot.png](https://issues.jenkins.io/secure/attachment/27993/ScreenShot.png)
- [bugchugger_cleanup_1098516.txt](https://issues.jenkins.io/secure/attachment/27994/bugchugger_cleanup_1098516.txt)
- [bugchugger_cleanup_1098517.txt](https://issues.jenkins.io/secure/attachment/27995/bugchugger_cleanup_1098517.txt)
- [bugchugger_cleanup_1098518.txt](https://issues.jenkins.io/secure/attachment/27996/bugchugger_cleanup_1098518.txt)
- [error](https://issues.jenkins.io/secure/attachment/27997/error)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rdrabens's <a href="https://issues.jenkins.io/browse/JENKINS-11962?focusedId=200357&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200357">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I am also seeing this issue (version 1.451).  I have up to 7 jobs running concurrently and I'm getting the "ln -s ... failed: 17 File exists" error message in about 5% of the jobs.  That means that there is almost always a group of jobs stuck because of a job that is unable to successfully create the symlink.  The stuck jobs eventually finish, sometimes in a few seconds, sometimes up to 20 minutes later.  </p>
</pre>
</details>

</details>
<p>I am also seeing this issue (version 1.451).  I have up to 7 jobs running concurrently and I'm getting the "ln -s ... failed: 17 File exists" error message in about 5% of the jobs.  That means that there is almost always a group of jobs stuck because of a job that is unable to successfully create the symlink.  The stuck jobs eventually finish, sometimes in a few seconds, sometimes up to 20 minutes later.  </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-11962?focusedId=213698&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-213698">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Can this be reproduced in more recent versions of Jenkins (no older than 8-10 weeks or so)? If so, what OS and what version and vendor of Java are you using? Please include log excerpts, content of the /systemInfo URL in your comment. Also relevant:
<a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a></p>
</pre>
</details>

</details>
<p>Can this be reproduced in more recent versions of Jenkins (no older than 8-10 weeks or so)? If so, what OS and what version and vendor of Java are you using? Please include log excerpts, content of the /systemInfo URL in your comment. Also relevant:<br/>
<a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-11962?focusedId=216202&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-216202">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I think the Java 1.6 implementation needs JNA to work or something similar. Maybe try running Jenkins on Java 7.</p>

<p>Also, it doesn't look like you need the builds to run in parallel, as they are all done within milliseconds. Disabling that would likely prevent issues like that.</p>
</pre>
</details>

</details>
<p>I think the Java 1.6 implementation needs JNA to work or something similar. Maybe try running Jenkins on Java 7.</p>

<p>Also, it doesn't look like you need the builds to run in parallel, as they are all done within milliseconds. Disabling that would likely prevent issues like that.</p>
